barretenberg/cpp/src/barretenberg/benchmark/indexed_tree_bench/indexed_tree.bench.cpp

```diff
@@ -27,24 +27,79 @@ const size_t MAX_BATCH_SIZE = 64;
 template <typename TreeType> void add_values(TreeType& tree, const std::vector<NullifierLeafValue>& values)
 {
     Signal signal(1);
-    typename TreeType::AddCompletionCallback completion = [&](const auto&) -> void { signal.signal_level(0); };
+    bool success = true;
```

Comment:
How does the sequential insertion compare with the batch insertion?

The mutli-threaded batch insertion doesn't perform as well as I had hoped. It was much better when we used pedersen as a much greater portion of the time was spent hashing.

Reply:
As discussed. Looks like the multi-threaded performs a little better in most scenarios. I think we should wait until the compete use case of the world state has settled further and then optimise.